getting_started/cavs-boot/apollolake/apl-boot-rom.rst

```diff
@@ -0,0 +1,162 @@
+   :caption: APL ROM Status Codes
+
+   #define FSR_ROM_INIT                                    0x0
+   #define FSR_ROM_INIT_DONE                               0x1
+   #define FSR_ROM_CSE_MANIFEST_LOADED                     0x2
+   #define FSR_ROM_FW_MANIFEST_LOADED                      0x3
+   #define FSR_ROM_FW_FW_LOADED                            0x4
+   #define FSR_ROM_FW_ENTERED                              0x5
+   #define FSR_ROM_VERIFY_FEATURE_MASK                     0x6
+   #define FSR_ROM_GET_LOAD_OFFSET                         0x7
+   #define FSR_ROM_BASEFW_CSE_IMR_REQUEST                  0x10
+   #define FSR_ROM_BASEFW_CSE_IMR_GRANTED                  0x11
+   #define FSR_ROM_BASEFW_CSE_VALIDATE_IMAGE_REQUEST       0x12
+   #define FSR_ROM_BASEFW_CSE_IMAGE_VALIDATED              0x13
+
+.. code-block:: c
+   :caption: APL ROM Error Codes
+
+   #define ADSP_UNHANDLED_INTERRUPT                        0xBEE00000
+
+   // Memory hole/ECC error
+   // Status bits are provided:
+   // [0] - L2 SRAM ECC error
+   // [1] - L2 memory hole error
+   #define ADSP_MEMORY_HOLE_ECC                            0xECC00000
+   #define ADSP_USER_EXCEPTION                             0xBEEF0000
+   #define ADSP_KERNEL_EXCEPTION                           0xCAFE0000
+
+   // Other critical error
+   #define ADSP_FAILURE                                    6
+   // FW image does not match the feature mask read from HW register.
+   #define ADSP_INVALID_FEAT_MASK                          20
+   // Invalid parameter
+   #define ADSP_INVALID_PARAM                              21
+   // CSE responded with error on an IPC request
+   #define ADSP_CSE_ERROR                                  40
+   // Invalid IPC response sent back by CSE.
+   #define ADSP_CSE_WRONG_RESPONSE                         41
+   // Size of IMR assigned by CSE is too small to load FW Image.
+   #define ADSP_IMR_TOO_SMALL                              42
+   // Base FW module not found in FW Image.
+   #define ADSP_BASE_FW_NOT_FOUND                          43
+   // CSE responded with error on FW image validation request.
+   #define ADSP_CSE_VALIDATION_FAILED                      44
+   // IPC communication failed with fatal error.
+   #define ADSP_IPC_FATAL_ERROR                            45
+   // L2 cache command failed.
+   #define ADSP_L2_CACHE_ERROR                             46
+   // Load offset set in FW Image Manifest is too small.
+   #define ADSP_LOAD_OFFSET_TOO_SMALL                      47
+
+ROM -> FW Transition
+====================
+
+Once APL ROM jumps to the entry point of the first module in the main binary,
+the memory and caches are in the following state:
+
+* L2$ is turned on, so the FW boot procedure may either execute via L2
+  cacheable address space or directly via L2 uncacheable alias.
+
+* HPSRAM areas allocated by the ROM listed in the next table.
+
+APL ROM HPSRAM Allocation
+=========================
+
++---------------------+------------+--------------+
```

Comment:
If you find creating table like this is hard to maintain, there Sphinx directives to make creating tables easier (check out the csv-table and list-table directives (with examples here: http://sublime-and-sphinx-guide.readthedocs.io/en/latest/tables.html )